src/lib/images/proxy.ts

```diff
@@ -152,7 +152,32 @@ export const initializeProxy = async function ({
       console.log(`fullImageUrl: ${fullImageUrl}`)
       req.url = `/${modifiers}/${fullImageUrl}`
     } else {
-      // If the image is remote, we can just pass the URL as is
+      // If the image is remote, we first check if it's allowed by any of patterns
+      if (!remoteImages.some((remoteImage) => remoteImage.test(sourceImagePath))) {
+        const remoteImageNotAllowedLogMessage = `Remote image "${sourceImagePath}" source for Image CDN is not allowed.`
+
+        // Contextual information about the remote image configuration is throttled
+        // to avoid spamming the console as it's quite verbose
+        // Each not allowed remote image will still be logged, just without configuration details
+        if (Date.now() - lastTimeRemoteImagesConfigurationDetailsMessageWasLogged > 1000 * 30) {
+          console.error(
+            `${remoteImageNotAllowedLogMessage}\n\n${
+              remoteImages.length === 0
+                ? 'Currently no remote images are allowed.'
+                : `Currently allowed remote images configuration details:\n${remoteImages
+                    .map((pattern) => ` - ${pattern}`)
+                    .join('\n')}`
+            }\n\nSee https://docs.netlify.com/image-cdn/overview/#remote-path how to configure allowed remote images.`,
+          )
+          lastTimeRemoteImagesConfigurationDetailsMessageWasLogged = Date.now()
+        } else {
+          console.error(remoteImageNotAllowedLogMessage)
```

Comment:
Can you use the `log()` method instead, which respects the global verbosity settings?

https://github.com/netlify/cli/blob/e1372a3c228246c7feb9bf4f4e63dad547b2bfa9/src/utils/command-helpers.ts#L163